plugin/src/test/groovy/com/github/triplet/gradle/play/PlayPublishTaskTest.groovy

```diff
@@ -1,167 +0,0 @@
-            new Track().setTrack(TrackType.PRODUCTION.publishedName)
-    ]
-    final TracksListResponse tracksListResponse = new TracksListResponse().setTracks(tracksList)
-
-    @Before
-    void setup() {
-        initMocks(this)
-
-        appEdit.setId('424242')
-        apk.setVersionCode(42)
-
-        doReturn(editsMock).when(publisherMock).edits()
-        doReturn(insertMock).when(editsMock).insert(anyString(), nullable(AppEdit.class))
-        doReturn(appEdit).when(insertMock).execute()
-
-        doReturn(editTracksListMock).when(editTracksMock).list(anyString(), anyString())
-        doReturn(tracksListResponse).when(editTracksListMock).execute()
-
-        doReturn(apksMock).when(editsMock).apks()
-        doReturn(uploadMock).when(apksMock).upload(anyString(), anyString(), nullable(FileContent.class))
-        doReturn(apk).when(uploadMock).execute()
-
-        doReturn(editTracksMock).when(editsMock).tracks()
-        doReturn(getInternalTrackMock).when(editTracksMock).get(anyString(), anyString(), eq('internal'))
-        doReturn(internalTrack).when(getInternalTrackMock).execute()
-        doReturn(getAlphaTrackMock).when(editTracksMock).get(anyString(), anyString(), eq('alpha'))
-        doReturn(alphaTrack).when(getAlphaTrackMock).execute()
-        doReturn(getBetaTrackMock).when(editTracksMock).get(anyString(), anyString(), eq('beta'))
-        doReturn(betaTrack).when(getBetaTrackMock).execute()
-
-        doReturn(tracksUpdateMock).when(editTracksMock).update(anyString(), anyString(), anyString(), nullable(Track.class))
-        doReturn(editCommitMock).when(editsMock).commit(anyString(), anyString())
-    }
-
-    @Test
-    void testApplicationIdChange() {
-        def project = TestHelper.evaluatableProject()
-
-        project.android {
-            flavorDimensions 'pricing'
-
-            productFlavors {
-                paid {
-                    dimension 'pricing'
-                    applicationId 'com.example.publisher'
-                }
-                free {
-                    dimension 'pricing'
-                }
-            }
-
-            buildTypes {
-                release {
-                    applicationIdSuffix '.release'
-                }
-            }
-
-            applicationVariants.all { variant ->
-                def flavorName = variant.variantData.variantConfiguration.flavorName
-                if (flavorName == 'paid') {
-                    variant.mergedFlavor.applicationId += '.paid'
-                }
-            }
-        }
-
-        project.evaluate()
-
-        // Attach the mock
-        setMockPublisher(project.tasks.publishPaidReleaseApk)
-
-        // finally run the task we want to check
-        project.tasks.publishPaidReleaseApk.publishApks(inputsMock)
-
-        // verify that we init the connection with the correct application id
-        verify(editsMock).insert('com.example.publisher.paid.release', null)
```

Comment:
Why did we have to remove this test? This was part of a contribution in #55 to make sure we're deferring application id resolution to when the task is actually running (some people were running into problems because they did some applicationId magic as part of the build). I believe we still support that behavior because of all those changes regarding late initialization and configuration avoidance. I agree that this test is super heavy but I'd prefer not to delete tests if it's not absolutely necessary. Or let's at least explain why a test can be safely removed